pkg/webhook/server/server.go

```diff
@@ -343,26 +343,22 @@ func (s *Server) mutate(obj runtime.Object) (runtime.Object, error) {
 }
 
 func (s *Server) convert(obj runtime.Object) (runtime.Object, error) {
-	outputVersion := apiextensionsv1.SchemeGroupVersion
-	review, isV1 := obj.(*apiextensionsv1.ConversionReview)
-	if !isV1 {
-		outputVersion = apiextensionsv1beta1.SchemeGroupVersion
-		reviewv1beta1, isV1beta1 := obj.(*apiextensionsv1beta1.ConversionReview)
-		if !isV1beta1 {
-			return nil, errors.New("request is not of type apiextensions v1 or v1beta1")
+	switch review := obj.(type) {
+	case *apiextensionsv1.ConversionReview:
+		if review.Request == nil {
+			return nil, errors.New("review.request was nil")
 		}
-		convertedReview, err := defaultScheme.ConvertToVersion(reviewv1beta1, apiextensionsv1.SchemeGroupVersion)
-		if err != nil {
-			return nil, err
+		review.Response = s.ConversionWebhook.ConvertV1(review.Request)
+		return review, nil
+	case *apiextensionsv1beta1.ConversionReview:
+		if review.Request == nil {
+			return nil, errors.New("review.request was nil")
 		}
-		review = convertedReview.(*apiextensionsv1.ConversionReview)
+		review.Response = s.ConversionWebhook.ConvertV1Beta1(review.Request)
+		return review, nil
+	default:
+		return nil, fmt.Errorf("unsupported conversion review type: %T", review)
```

Comment:
Rather than attempt to convert between the ConversionReview versions, I have instead dispatched to two version specific Convert methods on the ConversionWebhook....somewhat inspired by https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/webhook.go